python/cudf/cudf/core/index.py

```diff
@@ -2661,6 +2664,103 @@ def categories(self):
         return self._values.cat().categories
 
 
+class interval_range(GenericIndex):
+    """
+    Returns a fixed frequency IntervalIndex.
+
+    Parameters
+    start : numeric, default None
+    Left bound for generating intervals.
+
+    end : numeric , default None
+    Right bound for generating intervals.
+
+    freq : numeric, default None
+    The length of each interval. Must be consistent
+    with the type of start and end
+
+    name : str, default None
+    Name of the resulting IntervalIndex.
+
+    closed : {‘left’, ‘right’, ‘both’, ‘neither’}, default ‘right’
+    Whether the intervals are closed on the left-side, right-side,
+    both or neither.
+
+    Returns
+    IntervalIndex
+    """
+
+    def __new__(
+        cls, start=None, end=None, freq=None, closed="right", name=None,
+    ) -> "interval_range":
+        if freq:
+            freq = freq  # freq if other than 1
+        else:
+            freq = 1
+        sf = start  # start+freq
+        ef = end + 1  # end + 1
+
+        arr = pd.arrays.IntervalArray.from_breaks(
+            [i for i in range(sf, ef, freq)], closed=closed
+        )
```

Comment:
I think we can do this on the GPU without Pandas using an approach like the following:

1. Use . [`arange`](https://github.com/rapidsai/cudf/blob/branch-0.18/python/cudf/cudf/core/column/column.py#L2027) to generate two column representing the `left` and `right` values.

2. Construct an `IntervalColumn` from the `left` and `right` column. For this, we should build a utility called `build_interval_column` in `column.py`, which builds an `IntervalColumn` given a `left` column and a `right` column (similar to `build_categorical_column` which builds a `CategoricalColumn` given a `codes` column and a `categories` column).

3. Pass the resulting `IntervalColumn` to `IntervalIndex` below.